src/query/sharing/src/share_endpoint_client.rs

```diff
@@ -167,4 +168,41 @@ impl ShareEndpointClient {
             }
         }
     }
+
+    #[async_backtrace::framed]
+    pub async fn get_reference_table_by_name(
+        &self,
+        share_params: &ShareDBParams,
+        from_tenant: &str,
+        db: &str,
+        table_name: &str,
+    ) -> Result<TableInfo> {
+        let to_tenant = share_params.share_ident.tenant_name();
+        let share_name = share_params.share_ident.share_name();
+        let path = format!(
+            "/{}/{}/{}/{}/v2/reference_table",
+            to_tenant, share_name, db, table_name
+        );
+        // skip path first `/` char
+        let uri = format!("{}{}", share_params.share_endpoint_url, &path[1..]);
+        let headers = Self::generate_auth_headers(
+            &path,
+            &share_params.share_endpoint_credential,
+            from_tenant,
+        );
+        let client = reqwest::Client::new();
```

Comment:
Creating a client each time prevents us from reusing the same TCP connections.

Reply:
OK, after merge this pr, i will start a new optimize pr to improve performance, including:
* reuse http client;
* global presigned cache
* remove redundancy share endpoint request
* ...

in this optimize pr, i will give new benchmark results.